Src/SDK/WorkflowClient/Temporal.WorkflowClient/internal/TemporalServiceInvoker.cs

```diff
@@ -195,39 +193,145 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
+        public async Task<StartWorkflow.Result> StartWorkflowWithSignalAsync<TWfArg, TSigArg>(StartWorkflow.Arguments.WithSignal<TWfArg, TSigArg> opArgs)
+        {
+            // We need to re-validate the arguments because they went through the interceptor pipeline and thus may have
+            // been modified by customer code.

```

Comment:
It's open source, this isn't just for "customers"

```suggestion
            // been modified by user code.

```

Reply:
✔️